File: RootTools2/src/main/java/com/rarnu/tools/neo/base/InnerPreferenceFragment.java
package com.rarnu.tools.neo.base;

import android.os.Bundle;
import android.preference.PreferenceFragment;
import android.view.Menu;
import android.view.MenuInflater;

public abstract class InnerPreferenceFragment extends PreferenceFragment implements IIntf {

    public String tabTitle = null;
    protected Bundle innerBundle = null;

    public InnerPreferenceFragment() {
        super();
    }
    public InnerPreferenceFragment(String tabTitle) {
        super();
        this.tabTitle = tabTitle;
    }

    @Override
    public void onActivityCreated(Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        innerBundle = getArguments();
        initComponents();
        initEvents();
        initLogic();
    }

    public void setNewArguments(Bundle bn) {
        innerBundle = bn;
        onGetNewArguments(bn);
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setHasOptionsMenu(true);
        addPreferencesFromResource(getFragmentLayoutResId());
        if (getActivity().getActionBar() != null) {
            if (getCustomTitle() == null || getCustomTitle().equals("")) {
                getActivity().getActionBar().setTitle(getBarTitle());
            } else {
                getActivity().getActionBar().setTitle(getCustomTitle());
            }
        }
    }

    @Override
    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
        if (getActivity() == null) {
            return;
        }
        initMenu(menu);
    }

}

File: RootTools2/src/main/kotlin/com/rarnu/tools/neo/base/InnerPreferenceFragment.kt
package com.rarnu.tools.neo.base

import android.content.Context
import android.os.Build
import android.os.Bundle
import android.preference.PreferenceFragment
import android.view.Menu
import android.view.MenuInflater

/**
 * Created by rarnu on 3/23/16.
 */
abstract class InnerPreferenceFragment : PreferenceFragment, IIntf {

    var tabTitle: String? = null
    protected var innerBundle: Bundle? = null

    constructor() : super()
    constructor(tabTitle: String) : super() {
        this.tabTitle = tabTitle
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        innerBundle = arguments
        initComponents()
        initEvents()
        initLogic()
    }

    open fun setNewArguments(bn: Bundle?) {
        innerBundle = arguments
        onGetNewArguments(bn)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setHasOptionsMenu(true)
        addPreferencesFromResource(getFragmentLayoutResId())

        if (activity.actionBar != null) {
            if (getCustomTitle() == null || getCustomTitle() == "") {
                activity.actionBar.setTitle(getBarTitle())
            } else {
                activity.actionBar.title = getCustomTitle()
            }
        }
    }

    override fun onCreateOptionsMenu(menu: Menu?, inflater: MenuInflater?) {
        if (activity == null) {
            return
        }
        initMenu(menu)
    }

    override fun getContext(): Context? {
        if (Build.VERSION.SDK_INT >= 23) {
            return super.getContext()
        } else {
            return super.getActivity()
        }
    }
}